### PR TITLE
Conda outdated package helper

### PR DESCRIPTION
**Pull request by romainx**
Hello,

I propose to integrate a feature--still experimental at this time-- in order to help identifying which packages can be updated. It could be useful in the frame of maintaining stacks up to date (see #1014). It could be part of the [Package Update documentation](https://jupyter-docker-stacks.readthedocs.io/en/latest/contributing/packages.html) documentation.

It's implemented as a test that is run through a specific command. It's skipped by default when running standard tests through `make test/<stack>`--however I can be integrated if we think it could be useful in the test context.

The principle is to call a specific entry `make check_outdated/<stack>` in the `Makefile` to get a report of the packages that can be updated.

```bash
$ make check_outdated/base-notebook

# 2020-02-12 20:13:51 [    INFO] 27 packages that can be updated over 110 -> 25% (test_outdated.py:76)
# 2020-02-12 20:13:51 [    INFO] 
# Package            Current     Newest
# -----------------  ----------  --------
# alembic            1.3.3       1.4.0
# ca-certificates    2019.11.28  2020.1.1
# cffi               1.13.2      1.14.0
# conda              4.7.12      4.8.2
# inflect            4.0.0       4.1.0
# ipython            7.11.1      7.12.0
# jinja2             2.11.0      2.11.1
# json5              0.8.5       0.9.1
# jupyter_core       4.6.1       4.6.2
# jupyter_telemetry  0.0.4       0.0.5
# jupyterlab         1.2.5       1.2.6
# krb5               1.16.4      1.17.1
# libcurl            7.65.3      7.68.0
# libsodium          1.0.17      1.0.18
# nodejs             13.7.0      13.8.0
# [...]                                                                                                                                                                                                 
```

It's a very first version, please give me your opinion!

PS: @parente here is the closed pull request #1008 .

## Notes 

- It's heavily based on the script provided in [this](https://oerpli.github.io/post/2019/06/conda-outdated/) blog post. I've kept the copyright notice but I'm not confortable with that.
- It uses [tabulate](https://github.com/astanin/python-tabulate) to display the table in the report. The dependency has been added in the `requirements-dev.txt`.

### CODE REVIEW

**Comment by romainx** [comment]
Flagged as **WIP** since I'm working on an improvement to filter the list in order to keep only packages requested (specs). So the list of outdated packages will not contain anymore the dependencies since we do not master them but only the packages specified in the `Dockerfiles`.
Do not hesitate to give your feedback 😄

**Comment by romainx** [comment]
Version improved! 
I keep only packages that have been installed explicitly in the `Dockerfiles`.
Here is an example on the `scipy-notebook` cf. #1014 . So you know exactly what to do.

```bash
$ make check-outdated/scipy-notebook

# [...]
# INFO     test_outdated:test_outdated.py:80 9/33 (27%) packages could be updated
# INFO     test_outdated:test_outdated.py:82 
# Package      Current    Newest
# -----------  ---------  --------
# cloudpickle  1.2.2      1.3.0
# conda        4.7.12     4.8.2
# dask         2.9.2      2.11.0
# hdf5         1.10.5     1.10.6
# jupyterlab   1.2.5      1.2.6
# pandas       0.25.3     1.0.1
# python       3.7.4      3.8.2
# seaborn      0.9.0      0.10.0
# statsmodels  0.11.0     0.11.1
```

**Comment by romainx** [comment]
Working on it to clean it up, performing some refactoring in the aim of making it more reusable.

**Comment by romainx** [comment]
@parente All good, I've just finished to refactor the code. I intend to reuse it to perform other tests. I would be interested to have your opinion :smile:.

**Comment by romainx** [comment]
Sorry I have just noticed that an option exist that could make the code much simpler. I will use it instead of performing a custom parsing.

```bash
conda env export -n base --from-history --json
```

File: test/helpers.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+# CondaPackageHelper is partially based on the work https://oerpli.github.io/post/2019/06/conda-outdated/.
+# See copyright below.
+#
+# MIT License
+# Copyright (c) 2019 Abraham Hinteregger
```
Comment (parente) [inline test/helpers.py]: 👍

**Pull request merged**